setup/setup_snyk.sh

```diff
@@ -63,11 +57,40 @@ fi
 
 chmod +x snyk
 ${SUDO_CMD} mv snyk /usr/local/bin
+# Function to download a file with fallback to backup URL
+# Parameters:
+#   $1: File name to download
+#   $2: Output file name
+download_file() {
+    # Try to download from the main URL
+    if curl --compressed --retry 2 --output "$2" "$MAIN_URL/$1?utm_source="$GH_ACTIONS; then
+        echo "Downloaded $1 from main URL"
```

Comment:
nitpick: helpful logs are important, so why not just print the actual URL?

```suggestion
        echo "Downloaded from $MAIN_URL/$1"
```